src/core/dbus-service.c

```diff
@@ -171,6 +171,9 @@ static int bus_set_transient_exit_status(
         if (r < 0)
                 return r;
 
+        sz_status /= sizeof(int);
+        sz_signal /= sizeof(int);
```

Comment:
needs to be int32_t (the current version of this function is borked, see earlier comment)